pycbc/events/eventmgr.py

```diff
@@ -347,18 +371,25 @@ def add_template_events(self, columns, vectors):
                     new_events[c] = v.numpy()
                 else:
                     new_events[c] = v
-        self.template_events = numpy.append(self.template_events, new_events)
+        new_size = self.template_event_size+len(new_events)
+        if new_size > len(self.template_events):
+            self.template_events.resize(
+                new_size + self.array_minsize,
+                refcheck=False
+            )
+        self.template_events[self.template_event_size:new_size] = new_events
+        self.template_event_size += len(new_events)
 
     def cluster_template_events(self, tcolumn, column, window_size):
         """ Cluster the internal events over the named column
         """
-        cvec = self.template_events[column]
-        tvec = self.template_events[tcolumn]
+        cvec = self.template_events[column][:self.template_event_size]
+        tvec = self.template_events[tcolumn][:self.template_event_size]
         if window_size == 0:
             indices = numpy.arange(len(tvec))
         else:
             indices = findchirp_cluster_over_window(tvec, cvec, window_size)
-        self.template_events = numpy.take(self.template_events, indices)
+        self.template_events[:len(indices)] = self.template_events[indices]
```

Comment:
Do you not need to update `self.template_event_size` after this?

Also, if `window_size` is zero, the function is effectively a no-op, so the function can be considerably simplified by moving the check at the very beginning and just returning.